crates/ty_ide/src/semantic_tokens.rs

```diff
@@ -0,0 +1,2531 @@
+
+                // Visit decorator expressions
+                self.visit_decorators(&func.decorator_list);
+
+                // Clear the in_class_scope flag so inner functions
+                // are not treated as methods
+                let prev_in_class = self.in_class_scope;
+                self.in_class_scope = false;
+                walk_stmt(self, stmt);
+                self.in_class_scope = prev_in_class;
+            }
+            ast::Stmt::ClassDef(class) => {
+                // Class name
+                self.add_token(
+                    class.name.range(),
+                    SemanticTokenType::Class,
+                    vec![SemanticTokenModifier::Definition],
+                );
+
+                // Type parameters (Python 3.12+ syntax)
+                if let Some(type_params) = &class.type_params {
+                    self.visit_type_params(type_params);
+                }
+
+                // Handle base classes and type annotations in inheritance
+                if let Some(arguments) = &class.arguments {
+                    // Visit base class arguments
+                    for arg in &arguments.args {
+                        self.visit_expr(arg);
+                    }
+                    // Visit keyword arguments (for metaclass, etc.)
+                    for keyword in &arguments.keywords {
+                        self.visit_expr(&keyword.value);
+                    }
+                }
+
+                // Visit decorator expressions
+                self.visit_decorators(&class.decorator_list);
+
+                let prev_in_class = self.in_class_scope;
+                self.in_class_scope = true;
+                walk_stmt(self, stmt);
+                self.in_class_scope = prev_in_class;
+            }
+            ast::Stmt::AnnAssign(assign) => {
+                // Handle annotated assignments (e.g., x: int = 5)
+                if let ast::Expr::Name(name) = assign.target.as_ref() {
+                    let (token_type, modifiers) = self.classify_name(name);
+                    self.add_token(name.range(), token_type, modifiers);
+                }
+
+                // Handle the type annotation
+                self.visit_type_annotation(&assign.annotation);
+
+                // Handle the value if present
+                if let Some(value) = &assign.value {
+                    self.visit_expr(value);
+                }
+            }
+            ast::Stmt::Import(import) => {
+                for alias in &import.names {
+                    if let Some(asname) = &alias.asname {
+                        self.add_token(asname.range(), SemanticTokenType::Namespace, vec![]);
+                    } else {
+                        // Create separate tokens for each part of a dotted module name
+                        self.add_dotted_name_tokens(&alias.name, SemanticTokenType::Namespace);
+                    }
+                }
+                walk_stmt(self, stmt);
```

Comment:
This `walk_stmt` call seems redundant as visiting an import statement in source order is only about visiting the aliases which is already happening above.